### DISCUSSION TITLE
invalid syntax error at import microdot  in in micropython

### DISCUSSION THREAD

**Discussion by proximacentaury** [miguelgrinberg/microdot#220] [Q&A] [answered]
Hi,

Sorry about that beginner question but I am stuck completelly.
I am new with micropython, and microdot on ESP32.
The micropython installed well, the webrepl working fine.
If I make some simple blinking example everyithing is fine.
My problem starts with
` from microdot import Microdot`
Everything looks fine, but after the reboot I have receive the following:
```
Traceback (most recent call last):
  File "main.py", line 1, in <module>
  File "microdot.py", line 7
SyntaxError: invalid syntax
MicroPython v1.22.2 on 2024-02-22; Generic ESP32 module with ESP32
Type "help()" for more information.
```
I have started with Pycharm IDE, but I have same problem with Thonny as well. (As mentioned before I am new with programming, I and I am not sure even with IDEs neither.
I have tried different version of microdot as well without any success,

I am not able to discover the problem right now.
Any help thank you in advance

**Comment by miguelgrinberg** [comment, marked as answer]
Can you show me the first 10 lines of microdot.py? The error appears in line 7, but the official microdot.py file has a comment in that line, so you seem to have a modified file.

**Reply by proximacentaury** [reply]
Thank you for the answer. Finally I have found the problem based on idea.

**Comment by proximacentaury** [comment]
Hello Miquel,

I am sorry for the late answer because I have spent some time to  with the problem.
Fortunatelly I found the reason of the invalid syntax problem...
There was a mistake. Somehow I uploaded  the html file of the github page except the microdot itself.... :-)
The Tonny is very usefull, becuse it is able the load "back" the file from ESP32 to computer, just see what was really uploaded.
But on the other hand I am not able to make a running up.
Can you sugest a minimalistic concrete solid example just to test the setup?

Thank you in advance...

**Reply by miguelgrinberg** [reply]
Try [this one](https://github.com/miguelgrinberg/microdot/blob/main/examples/hello/hello.py), which is the simple example. Then you can try other examples, there are [quite a few](https://github.com/miguelgrinberg/microdot/blob/main/examples)

**Reply by proximacentaury** [reply]
Hello Miquel,

This is an answer what I was affraid of... :-)
I have tried the mentioned files without success...
Now I am trying with an ESP8266 as well.
The result is very strange. 
`Traceback (most recent call last):
  File "main.py", line 1, in <module>
MemoryError: 
`
The main.py is the microdot sample:


`from microdot import Microdot

app = Microdot()

@app.route('/')
async def index(request):
    return 'Hello, world!'

app.run()`

The error is different, but the location of the error at the import again!
I have found some forum posts around the micropython, where the TRY  and function "hides" the real reason of the error.  Which could be a simple typo.

I am pretty sure I making a very basic mistake, but I have no knowledge to discover the problem....

**Reply by miguelgrinberg** [reply]
It's a memory error. Have you read the section of the documentation regarding pre-compiling your files? The 8266 is a very limited device, maybe the compiler runs out of memory when importing microdot.py. Try pre-compiling on your computer, hopefully that will help.
